server/models/meshmodel/core/register.go

```diff
@@ -1,65 +1,282 @@
 package core
 
 import (
+	"context"
 	"encoding/json"
 	"fmt"
+	"strings"
 
+	"cuelang.org/go/cue"
+	"cuelang.org/go/cue/cuecontext"
+	cueJson "cuelang.org/go/encoding/json"
+	"github.com/layer5io/meshkit/models/meshmodel"
 	"github.com/layer5io/meshkit/models/meshmodel/core/v1alpha1"
-
-	"github.com/layer5io/meshery/server/internal/store"
+	oamcore "github.com/layer5io/meshkit/models/oam/core/v1alpha1"
+	"github.com/layer5io/meshkit/utils/kubernetes"
+	"github.com/layer5io/meshkit/utils/manifests"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-type capability struct {
-	ID string `json:"id,omitempty"`
-	// Host is the address of the service registering the capability
-	Host string `json:"host,omitempty"`
+type crd struct {
+	Items []crdhelper `json:"items"`
 }
-
-// SetID sets the ID of the capability
-func (cap *capability) SetID(id string) {
-	cap.ID = id
+type crdhelper struct {
+	Spec spec `json:"spec"`
+}
+type spec struct {
+	Names names `json:"names"`
+}
+type names struct {
+	Kind string `json:"kind"`
 }
 
-// GetID returns the ID of the capability
-func (cap *capability) GetID() string {
-	return cap.ID
+func RegisterMeshmodelComponentsForCRDS(reg meshmodel.RegistryManager, k8sYaml []byte, contextID string, version string) {
+	//TODO: Replace GenerateComponents in meshkit to natively produce MeshModel components to avoid any interconversion
+	comp, err := manifests.GenerateComponents(context.Background(), string(k8sYaml), manifests.K8s, manifests.Config{
+		CrdFilter: manifests.NewCueCrdFilter(manifests.ExtractorPaths{
+			NamePath:    "spec.names.kind",
+			IdPath:      "spec.names.kind",
+			VersionPath: "spec.versions[0].name",
+			GroupPath:   "spec.group",
+			SpecPath:    "spec.versions[0].schema.openAPIV3Schema.properties.spec"}, false),
+		ExtractCrds: func(manifest string) []string {
+			crds := strings.Split(manifest, "---")
+			return crds
+		},
+	})
+	if err != nil {
+		fmt.Println("err: ", err.Error())
+		return
+	}
+	for i, schema := range comp.Schemas {
+		var def oamcore.WorkloadDefinition
+		err := json.Unmarshal([]byte(comp.Definitions[i]), &def)
+		if err != nil {
+			fmt.Println("err here: ", err.Error())
+			return
+		}
+		_ = reg.RegisterEntity(meshmodel.Host{
+			Hostname:  "kubernetes",
+			ContextID: contextID,
+		}, v1alpha1.ComponentDefinition{
+			Schema: schema,
+			TypeMeta: v1alpha1.TypeMeta{
+				APIVersion: def.Spec.Metadata["k8sKind"],
+				Kind:       def.Spec.Metadata["k8sAPIVersion"],
+			},
+			Metadata: v1alpha1.ComponentMetadata{
+				Model:   "kubernete",
```

Comment:
"kubernete" on purpose?

Reply:
oh geez, how do I make such silly mistakes. Thank you for your eyes.

Reply:
Made the change